src/components/command-bar/commands/search/hooks/use-recent-searches.ts

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useState } from 'react'
+
+const RECENT_SEARCHES_KEY = 'DEV_DOT_RECENT_SEARCHES'
+
+/**
+ * Get the current list of recent searches stored in local storage. If one does
+ * not exist, an empty one is created.
+ */
+const initializeRecentSearches = () => {
+	try {
+		const storageValue = window.localStorage.getItem(RECENT_SEARCHES_KEY)
+
+		if (!storageValue) {
+			window.localStorage.setItem(RECENT_SEARCHES_KEY, JSON.stringify([]))
+			return []
+		} else {
+			return JSON.parse(storageValue)
+		}
+	} catch (e) {
+		// window may be 'undefined'
+		// localStorage may be disabled
```

Comment:
⛏️ nit: We could `return []` here and simplify the check in the recent searches component.

```suggestion
		// localStorage may be disabled
		return []
```